src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const act = actions[i];
+
+    if (act.type === 'addProperties') {
+      Object.assign(stateCopy, act.extraData);
+    }
+
+    if (act.type === 'removeProperties') {
+      for (let j = 0; j < act.keysToRemove.length; j++) {
```

Comment:
better use `for .. of` [loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) here